boa3/model/builtin/classmethod/countmethod.py

```diff
@@ -0,0 +1,146 @@
+from typing import Any, Dict, List, Optional, Tuple
+
+from boa3.model.builtin.method.builtinmethod import IBuiltinMethod
+from boa3.model.expression import IExpression
+from boa3.model.type.collection.sequence.sequencetype import SequenceType
+from boa3.model.type.itype import IType
+from boa3.model.variable import Variable
+from boa3.neo.vm.opcode.Opcode import Opcode
+
+
+class CountMethod(IBuiltinMethod):
+
+    def __init__(self, sequence_type: Optional[SequenceType] = None, arg_value: Optional[IType] = None):
+        from boa3.model.type.type import Type
+        identifier = 'count'
+        if not isinstance(sequence_type, SequenceType):
+            sequence_type = Type.sequence
+        if not isinstance(arg_value, IType):
+            arg_value = Type.any
+
+        args: Dict[str, Variable] = {
+            'self': Variable(sequence_type),
+            'value': Variable(arg_value)
+        }
+        super().__init__(identifier, args, return_type=Type.int)
+
+    @property
+    def _arg_self(self) -> Variable:
+        return self.args['self']
+
+    def validate_parameters(self, *params: IExpression) -> bool:
+        if len(params) != 2:
+            return False
+        if not all(isinstance(param, IExpression) for param in params):
+            return False
+
+        from boa3.model.type.itype import IType
+        sequence_type: IType = params[0].type
+
+        if not isinstance(sequence_type, SequenceType):
+            return False
+        from boa3.model.type.collection.sequence.mutable.listtype import ListType
+        from boa3.model.type.collection.sequence.tupletype import TupleType
+        from boa3.model.type.collection.sequence.rangetype import RangeType
+        if not (isinstance(sequence_type, ListType) or isinstance(sequence_type, TupleType) or isinstance(sequence_type, RangeType)):
```

Comment:
you can test this with one `isinstance` check
```suggestion
        if not isinstance(sequence_type, (ListType, TupleType, RangeType)):
```